### PR TITLE
version sub-command added. [API-1456]

### PR DESCRIPTION
**Pull request by mehmettokgoz**
CLC prints followings when `version` command called:
- Command Line Client Version
- Latest Git commit hash
- Hazelcast Go Client Version
- Go Version

The CLC version is hard coded inside the `internal` package, similar to Go Client. Others are aimed to change dynamically at build and runtime. Since we created a new command, a new folder (`versioncmd`) has been created to accomplish `subCommand` pattern.

### CODE REVIEW

**Comment by mehmettokgoz** [comment]
Converted CLC version to change with every release.

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,10 @@
 .PHONY: build generate-completion test test-cover view-cover
 
 TAG=$(shell git describe --tags 2> /dev/null || echo unknown)
+GIT_COMMIT=$(shell git rev-parse HEAD 2> /dev/null || echo unknown)
+CLC_VERSION=$(shell git tag | sort -V | tail -1 2> /dev/null || echo unknown)
 CLIENT_TYPE="CLC"
-LDFLAGS="-X 'github.com/hazelcast/hazelcast-go-client/internal.ClientType=$(CLIENT_TYPE)' -X 'github.com/hazelcast/hazelcast-go-client/internal.ClientVersion=$(TAG)'"
+LDFLAGS="-X 'github.com/hazelcast/hazelcast-go-client/internal.ClientType=$(CLIENT_TYPE)' -X 'github.com/hazelcast/hazelcast-commandline-client/internal.GitCommit=$(GIT_COMMIT)' -X 'github.com/hazelcast/hazelcast-go-client/internal.ClientVersion=$(TAG)' -X 'github.com/hazelcast/hazelcast-commandline-client/internal.ClientVersion=$(CLC_VERSION)'"
```
Comment (yuce) [inline Makefile:7]: Let's use the same version number for both `github.com/hazelcast/hazelcast-go-client/internal.ClientVersion` and `github.com/hazelcast/hazelcast-commandline-client/internal.ClientVersion`

Comment (yuce) [reply Makefile:7]: Could you also remove `CLIENT_TYPE="CLC"` line and use `CLC` directly at `github.com/hazelcast/hazelcast-go-client/internal.ClientType=`?

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,10 @@
 .PHONY: build generate-completion test test-cover view-cover
 
 TAG=$(shell git describe --tags 2> /dev/null || echo unknown)
```
Comment (yuce) [inline Makefile:3]: Could you remove the `TAG` varible.

File: Makefile [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+GIT_COMMIT=$(shell git rev-parse HEAD 2> /dev/null || echo unknown)
+CLC_VERSION=$(shell git tag | sort -V | tail -1 2> /dev/null || echo unknown)
```
Comment (yuce) [inline Makefile:5]: Could you change `CLC_VERSION` to:
```
git describe --tags $(git rev-list --tags --max-count=1) || echo UNKNOWN
```

**Review by yuce — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (mehmettokgoz) [reply Makefile:5]: But this command gives a hash value, not the latest release tag. 

For example: `b11c5bdef4b055b2a28c1d1895b8d07ae990d2ec-1-g33465e9`

File: Makefile (hunk 1 above)
Comment (mehmettokgoz) [reply Makefile:7]: For the first comment, should we remove `github.com/hazelcast/hazelcast-commandline-client/internal.ClientVersion`and just use version number of Go Client which is `1.3.0`?

Comment (yuce) [reply Makefile:7]: We override the Go client version with CLC version in order to be able to show the correct version on MC, so we should use the CLC version, not Go Client version.

File: Makefile (hunk 3 above)
Comment (yuce) [reply Makefile:5]: It outputs this for me:
```
$ git describe --tags $(git rev-list --tags --max-count=1) || echo UNKNOWN
v1.0.0-beta1
```
Maybe you didn't fetch the tags?

File: Makefile (hunk 2 above)
Comment (mehmettokgoz) [reply Makefile]: Fixed at https://github.com/hazelcast/hazelcast-commandline-client/pull/107/commits/4d6e7149dd84650c120e2cd2d9323b80087d1ad9.

File: Makefile (hunk 1 above)
Comment (mehmettokgoz) [reply Makefile]: Fixed at https://github.com/hazelcast/hazelcast-commandline-client/pull/107/commits/4d6e7149dd84650c120e2cd2d9323b80087d1ad9.

**Review by yuce — APPROVED** [review]
Looks good!

**Pull request merged**